lib/clientv2/workloadendpoint_e2e_test.go

```diff
@@ -397,4 +410,106 @@ var _ = testutils.E2eDatastoreDescribe("WorkloadEndpoint tests", testutils.Datas
 			testWatcher4.Stop()
 		})
 	})
+	
+	Describe("WorkloadEndpoint prefix list", func() {
+		It("should handle prefix lists of workload endpoints", func() {
+			c, err := clientv2.New(config)
+			Expect(err).NotTo(HaveOccurred())
+
+			be, err := backend.NewClient(config)
+			Expect(err).NotTo(HaveOccurred())
+			be.Clean()
+
+			By("Creating two WorkloadEndpoint with same namespace, node, orchestrator and overlapping Pod")
+			outRes1, err := c.WorkloadEndpoints().Create(
+				ctx,
+				&apiv2.WorkloadEndpoint{
+					ObjectMeta: metav1.ObjectMeta{Namespace: "namespace1", Name: "node--1-k8s-pod-eth0"},
+					Spec:       apiv2.WorkloadEndpointSpec{
+						Node: "node-1",
+						Orchestrator: "k8s",
+						Pod: "pod",
+						Endpoint: "eth0",
+						InterfaceName: "cali1234",
+					},
+				},
+				options.SetOptions{},
+			)
+			Expect(err).NotTo(HaveOccurred())
+
+			outRes2, err := c.WorkloadEndpoints().Create(
+				ctx,
+				&apiv2.WorkloadEndpoint{
+					ObjectMeta: metav1.ObjectMeta{Namespace: "namespace1", Name: "node--1-k8s-pod--1-eth0"},
+					Spec:       apiv2.WorkloadEndpointSpec{
+						Node: "node-1",
+						Orchestrator: "k8s",
+						Pod: "pod-1",
+						Endpoint: "eth0",
+						InterfaceName: "cali1234",
+					},
+				},
+				options.SetOptions{},
+			)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Creating a workload in a different namespace, but with a largely overlapping name")
+			outRes3, err := c.WorkloadEndpoints().Create(
+				ctx,
+				&apiv2.WorkloadEndpoint{
+					ObjectMeta: metav1.ObjectMeta{Namespace: "namespace2", Name: "node--1-k8s-pod--2-eth0"},
+					Spec:       apiv2.WorkloadEndpointSpec{
+						Node: "node-1",
+						Orchestrator: "k8s",
+						Pod: "pod-2",
+						Endpoint: "eth0",
+						InterfaceName: "cali1235",
+					},
+				},
+				options.SetOptions{},
+			)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Doing an exact get on one of the workload endpoints")
+			outList, err := c.WorkloadEndpoints().List(ctx, options.ListOptions{Namespace: "namespace1", Name: "node--1-k8s-pod-eth0"})
+			Expect(err).NotTo(HaveOccurred())
+			Expect(outList.Items).To(HaveLen(1))
+			testutils.ExpectResource(&outList.Items[0], apiv2.KindWorkloadEndpoint, "namespace1", "node--1-k8s-pod-eth0", outRes1.Spec)
+
+			By("Doing a short prefix get to retrieve both workload endpoints in namespace1")
+			outList, err = c.WorkloadEndpoints().List(ctx, options.ListOptions{Namespace: "namespace1", Name: "node--1-k8s-pod-", Prefix: true})
+			Expect(err).NotTo(HaveOccurred())
+			Expect(outList.Items).To(HaveLen(2))
+			testutils.ExpectResource(&outList.Items[0], apiv2.KindWorkloadEndpoint, "namespace1", "node--1-k8s-pod--1-eth0", outRes2.Spec)
```

Comment:
is it always guaranteed to get the last one we put first in the list? Will this also be true for KDD?

Reply:
These test won't be valid for KDD since WEPs aren't createable.  I'll update to be etcdv3 only/

Reply:
The results will be returned by etcdv3 in alphanumeric order.